packages/cli/src/services/user.service.ts

```diff
@@ -276,12 +282,59 @@ export class UserService {
 		return await this.userRepository.manager.transaction(async (trx) => {
 			await trx.update(User, { id: user.id }, { role: { slug: newRole.newRoleName } });
 
-			const adminDowngradedToMember =
-				user.role.slug === 'global:admin' && newRole.newRoleName === 'global:member';
+			const isAdminRole = (roleName: string) => {
+				return roleName === 'global:admin' || roleName === 'global:owner';
+			};
+
+			const isDowngradedAdmin = isAdminRole(user.role.slug) && !isAdminRole(newRole.newRoleName);
 
-			if (adminDowngradedToMember) {
+			if (isDowngradedAdmin) {
```

Comment:
A downgraded chat user can still use their existing API keys to perform actions they shouldn't have access to?

Reply:
Great point, adjusting this.